pingcap/dm
# relay: retry for upstream bad connection

### What problem does this PR solve? 

- retry to read binlog events from upstream when encountering bad connection error

### What is changed and how it works?

- check return error from binlog reader
- use _pkg/backoff_ to trigger the retry

### Check List 

Tests 

 - Unit test
 - Manual test (add detailed scripts or steps below)
 	- run DM-worker with fetching binlog events
 	- `KILL` connection for the relay unit
 	- observe whether relay retry to fetch binlog events again

Side effects

 - Increased code complexity
